sdk/identity/azure-identity/tests/test_aad_client.py

```diff
@@ -304,7 +305,13 @@ def test_multitenant_cache():
     assert client_b.get_cached_access_token([scope]) is None
 
     # but C allows multitenant auth and should therefore return the token from tenant_a when appropriate
-    client_c = AadClient(tenant_id=tenant_c, **common_args)
+    client_c = AadClient(tenant_id=tenant_c, additionally_allowed_tenants=['*'], **common_args)
     assert client_c.get_cached_access_token([scope]) is None
     token = client_c.get_cached_access_token([scope], tenant_id=tenant_a)
     assert token.token == expected_token
+
+    # but d does not add target tenant into allowed list therefore fail
+    client_d = AadClient(tenant_id=tenant_d, **common_args)
+    assert client_d.get_cached_access_token([scope]) is None
+    with pytest.raises(ClientAuthenticationError):
```

Comment:
Can we also add an `assert`ion that the raised exception mentions "additionally_allowed_tenants"? Just to be sure that we preserve an informative exception message